### PR TITLE
Release GPU process memory after leaving for SD 1.5 and SD Turbo

### LINKED ISSUE #61: Memory is not released after leaving page for SD 1.5 and SD Turbo
The memory will keep increasing in following scenario:

### Steps:

1. Visit https://microsoft.github.io/webnn-developer-preview/demos/stable-diffusion-1.5/
2. Click "Load Models" button -> Wait for loading and compiling models completed

### Actual Result 1:

The memory of GPU process and Tab (blink process) will increase

Image

### Steps:

3. In the same tab, visit https://microsoft.github.io/webnn-developer-preview/demos/sd-turbo/

### Actual Result 2:

The memory of GPU process and Tab (blink process) from stable-diffusion-1.5 are not released

Image

### Steps:

4. Click "Load Models" button -> Wait for loading and compiling models completed

### Actual Result 3:

1. The memory of GPU process and Tab (blink process) keep increasing for both SD Turbo and SD 1.5
2. The probability of the "Context lost" error occurring is greatly increased, page blink and black screen happens

Image

### PR DESCRIPTION
**Pull request by ibelem**
Fix #61 

1. Add a "Release memory after leaving" switch in demo pages of SD 1.5 and SD Turbo, enabled by default
2. Run session.release() for browser `beforeunload` event

### When go to new page in same tab with "Release memory after leaving" switch disabled

1. The memories of GPU process and blink (Tab) process are still high
2. When click browser "Back" button, users can run "generate image" directly without loading models and compiling models

### When go to new page in same tab with "Release memory after leaving" switch enabled

1. The memory of GPU process will be released
2. The memory of blink (Tab) process is still high
3. When click browser "Back" button, users need to re-load and compile models then generate images

### Note

It seems like the ONNX Runtime Web cannot release memory of blink (Tab) process. 
- https://github.com/microsoft/onnxruntime/issues/18165 

>   @fs-eire WebAssembly memory cannot shrink. This means the wasm memory will keep the size of the peak memory usage, even if the code marks some part as "free".

- https://github.com/WebAssembly/memory-control/blob/main/proposals/memory-control/Overview.md 

@fdwr PTAL

CC @Honry @mingmingtasd @huningxin

### CODE REVIEW

File: demos/stable-diffusion-1.5/index.js [hunk 1]
```diff
@@ -325,41 +326,42 @@ async function readResponse(name, response) {
...
-const scLoad = document.querySelector("#scload");
-const scFetch = document.querySelector("#scfetch");
-const scCreate = document.querySelector("#sccreate");
-const scRun = document.querySelector("#scrun");
-const totalLoad = document.querySelector("#totalload");
-const totalRun = document.querySelector("#totalrun");
+const progressBarInner = $("#progress-bar-inner");
+const progressBarLabel = $("#progress-bar-label");
+const progressBarInnerInference = $("#progress-bar-inner-inference");
+const progressBarLabelInference = $("#progress-bar-label-inference");
```
Comment (fdwr) [inline demos/stable-diffusion-1.5/index.js:332]: Previously there was a call to `getElementById` and `querySelector` above, but the perlish-looking `$` now maps both to `document.querySelector`. Is that correct?

**Review by fdwr — APPROVED** [review]
Interesting - didn't know this memory was held. The `$` change adds a bunch of noise that would be nicer as a separate CR (or at least separate commit), but the actual change looks fine to me, assuming my one concern is reassured.

File: demos/stable-diffusion-1.5/index.js (hunk 1 above)
Comment (ibelem) [reply demos/stable-diffusion-1.5/index.js:332]: @fdwr getElementById() only works with ID attributes, while querySelector() can work with any CSS selector. 

`getElementById("<id>") == querySelector("#<id>").`

```
const $ = s => document.querySelector(s);
const $$ = s => [...document.querySelectorAll(s)];
```

Correct, $ maps to getElementById("<id>") and querySelector("#<id>").

**Comment by ibelem** [comment]
In the process of improving the memory code, I found that getElementById() and querySelector() were mixed used, and I couldn't help but change these codes. Will keep in mind the guidance of separating code formatting, refactoring, etc. from the original PR. :) @fdwr

**Pull request merged**